modules/library/main/src/main/java/org/geotools/feature/PropertyImpl.java

```diff
@@ -35,13 +35,11 @@ public abstract class PropertyImpl implements Property {
     /** descriptor of the property */
     protected PropertyDescriptor descriptor;
     /** user data */
-    protected final Map<Object, Object> userData;
+    private Map<Object, Object> _userData;
```

Comment:
Why the change on the access modifier and variable name? This class is not final and hence could have been extended by third-party extensions, this will break the backward compatibility.

Reply:
Good point, also, not following the adopted [coding convention](https://google.github.io/styleguide/javaguide.html#s5.1-identifier-names):

> In Google Style, special prefixes or suffixes are not used. For example, these names are not Google Style: name_, mName, s_name and kName.

Reply:
I think the reason for this change is that the field is now being lazily initialized. Therefore, subclasses should no longer use it directly but use the getter instead, otherwise they could get NPE. In that sense backwards compatibility for third-party subclasses is broken either way.

If we want the lazy initialization, this is the better option. Third parties will get a compilation fault when they upgrade geotools and are forced to fix it, in the other case they may not notice and get an NPE in production.

Reply:
Agreed on the lazy initialization changing the API anyways. Could not find users of the protected field outside of PropertyImpl. The coding convention bit still stands.

Reply:
Good point. Indeed it's due to the lazy initialization, which is beneficial because not all instances will have metadata. Re the variable name I can change it to something else that both adheres to the coding conventions and would result in a compile error of an eventual third party extension.

Reply:
Fixed: https://github.com/geotools/geotools/pull/4410/commits/45de0776540f133dae2073e270c22abc860f3756#diff-964e5fab63efd6e389520b1ecaf6b5bd87d769993a614c5568842437e90de9b4L37-L85